ssl/ssl_lib.c

```diff
@@ -4343,6 +4343,80 @@ const CTLOG_STORE *SSL_CTX_get0_ctlog_store(const SSL_CTX *ctx)
     return ctx->ctlog_store;
 }
 
+void
+SSL_CTX_set_early_cb(SSL_CTX *c, ssl_early_cb_fn cb, void *arg)
+{
+    c->early_cb = cb;
+    c->early_cb_arg = arg;
+}
+
+int
+SSL_EARLY_isv2(SSL *s, SSL_EARLY *ectx)
+{
+    return ectx->isv2;
+}
+
+unsigned int
+SSL_EARLY_get0_version(SSL *s, SSL_EARLY *ectx)
+{
+    return ectx->legacy_version;
+}
+
+size_t
+SSL_EARLY_get0_random(SSL *s, SSL_EARLY *ectx, const unsigned char **out)
+{
+    if (out != NULL)
+        *out = ectx->random;
+    return ectx->isv2 ? 0 : SSL3_RANDOM_SIZE;
```

Comment:
To be sure that someone who actually is familiar with the v2 protocol (i.e., not me) looks at this logic :)
I'm happy to take your word for it; it didn't really seem worth putting the time in to understand the v2-parsing code given that it's (hopefully) on its way out.